File: jasminesamples/lib/frequent_flyer.js
'use strict';

module.exports = (function () {

    var FrequentFlyer = function() {
        this.status = 'Bronze';
        this.statusPoints = 0;

        this.getStatus = function () {
            return this.status;
        };

        this.getStatusPoints = function() {
            return this.statusPoints;
        };

        this.earnStatusPoints = function(additionalPoints) {
            this.statusPoints = this.statusPoints + additionalPoints;
            if (this.statusPoints >= 300) {
                this.status = 'Silver';
            }
        }
    };

    return FrequentFlyer;
})();

File: rspec-examples/lib/frequent_flyer.rb
class FrequentFlyer
  def initialize()
    @status = 'BRONZE'
    @status_points = 0
  end

  def status
    @status
  end

  def status_points
    @status_points
  end

  def earn_status_points points
    @status_points = @status_points + points
    @status = 'SILVER' if status_points >= 300
    self
  end
end